### ISSUE TITLE
Timed Speed Limit

### ISSUE THREAD

**Issue by pcrosthwaite** [nzbget/nzbget#62]
Hi,

As a feature request, can you please look to add the ability to schedule how long a speed limit is in place for?

For example, when i click on the current speed to set a speed limit, have a drop down box with a number of timed options (10min, 30min, 1hr etc.). Once the time selected has passed, the speed limit is lifted and full speed is again achieved.

**Comment by hugbug** [comment]
I see how this can be useful. I have to find answers for few questions:
- when the time is up what speed to select: 0 (no limit), the previous limit, the limit as set in config file?
- what about active servers: does the timer resets only speed limit or active servers too (to what setting - see previous item)?
- how to indicate an active timer?

**Comment by pcrosthwaite** [comment]
Hi Andrey, When the time is up, it should revert back to the limit set in the config file, but if none was set, it should revert to 0.  I'm not sure with the active servers (I only have 1). Can you define different limits for different servers? If so, then it should revert back to the limit defined in the config file for that server, and back to 0 if none was defined.  As for indicating an active timer, the feature would need one of those clever icons, perhaps a snail? Either that or use the truck icon that currently indicates that a limit is in place, just with a count down beside the speed.  Thanks

---Sent from Boxer | http://getboxer.com

On 28 July 2015 4:34:03 am AEST, Andrey Prygunkov notifications@github.com wrote:I see how this can be useful. I have to find answers for few questions:   when the time is up what speed to select: 0 (no limit), the previous limit, the limit as set in config file?  what about active servers: does the timer resets only speed limit or active servers too (to what setting - see previous item)?  how to indicate an active timer?    —Reply to this email directly or view it on GitHub.

**Comment by hugbug** [comment]
> Can you define different limits for different servers?

No, servers can be only switched on/off. It seems logical to reset servers status as defined in config file.

I also wonder how well this function will play with "Pause for xx minutes". May be they should be combined somehow?

**Comment by pcrosthwaite** [comment]
Sabnzbd separates them, so you can either speed limit on a timer or pause on a timer. If you have both, I believe that pause trumps the speed limit. For example, if I have a speed limit active for 60min, I might then want to pause for 10min. In this case, the pause timer kicks in for 10min and once it expires, returns to the previous state, which in this case was a active speed limit (just with 50min left to run)  Hope this makes sense  

---Sent from Boxer | http://getboxer.com

On 28 July 2015 6:53:01 am AEST, Andrey Prygunkov notifications@github.com wrote: Can you define different limits for different servers?   No, servers can be only switched on/off. It seems logical to reset servers status as defined in config file.  I also wonder how well this function will play with "Pause for xx minutes". May be they should be combined somehow?  —Reply to this email directly or view it on GitHub.
